lib/proxy/log.rb

```diff
@@ -13,7 +20,18 @@ def logger
     def self.logger
       log_file = ::Proxy::SETTINGS.log_file
       if log_file.upcase == 'STDOUT'
+        if SETTINGS.daemon
+          puts "Settings log_file=STDOUT and daemon=true are incompatible, exiting..."
+          exit 1
+        end
         logger = ::Logger.new(STDOUT)
+      elsif log_file.upcase == 'SYSLOG'
+        begin
+          logger = ::Syslog::Logger.new 'foreman-proxy'
+        rescue
+          puts "Falling back to STDOUT logger"
+          logger = ::Logger.new(STDOUT)
+        end
       else
         # We keep the last 6 10MB log files
         logger = ::Logger.new(log_file, 6, 1024*1024*10)
```

Comment:
I agree, let me try to push fixed version (all your remarks above fixed) and we will see. I'd like to focus on the daemonizing part in this patch, touching this only because there is no way to log things while in daemon mode. That's the only reason why I include this work in the patch.